app/src/main/java/com/mybus/view/HistoryCardView.java

```diff
@@ -53,21 +53,18 @@ public void setList(List<RecentLocation> list) {
         mRecentLocationList = list;
         if (mRecentLocationList != null && mRecentLocationList.size() > 0) {
             mNoRecentTextView.setVisibility(GONE);
-            mHistoryItemsList.setVisibility(VISIBLE);
+            mRecyclerView.setVisibility(VISIBLE);
 
             // use this setting to improve performance if you know that changes
             // in content do not change the layout size of the RecyclerView
-            mHistoryItemsList.setHasFixedSize(true);
+            mRecyclerView.setHasFixedSize(true);
             // use a linear layout manager
             mLayoutManager = new LinearLayoutManager(getContext());
-            mHistoryItemsList.setLayoutManager(mLayoutManager);
+            mRecyclerView.setLayoutManager(mLayoutManager);
 
             mListAdapter = new HistoryItemAdapter(mRecentLocationList);
             mListAdapter.setItemSelectedListener(this);
-            mHistoryItemsList.setAdapter(mListAdapter);
-        } else {
-            mHistoryItemsList.setVisibility(GONE);
```

Comment:
Same as above, if you're setting the view a visibility, yo need to change it in the else, no matter on what state you setted it by the XML